app/src/main/kotlin/org/mifos/mobile/cn/ui/mifos/QRGenerator.kt

```diff
@@ -0,0 +1,59 @@
+package org.mifos.mobile.cn.ui.mifos
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.widget.Toast
+import com.google.gson.Gson
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel
+import kotlinx.android.synthetic.main.qrgenerator.*
+import org.mifos.mobile.cn.R
+import org.mifos.mobile.cn.ui.base.MifosBaseActivity
+
+class QRGenerator: MifosBaseActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.qrgenerator)
+        setToolbarTitle(getString(R.string.home))
+        getToolbar().setVisibility(View.GONE)
+        generateQrCodeButton.setOnClickListener {
+            if (checkEditText()) {
+                hideKeyboard()
+                val user = UserObject(name = fullNameEditText.text.toString(), mobile = mobileEditText.text.toString(),email = emailEditText.text.toString(),accountNumber = accountEditText.text.toString())
+                val serializeString = Gson().toJson(user)
+                setImageBitmap(serializeString)
+            }
+        }
+    }
+
+    private fun setImageBitmap(encryptedString: String?) {
+        val bitmap = QRCodeHelper.newInstance(this)?.setContent(encryptedString)?.setErrorCorrectionLevel(ErrorCorrectionLevel.Q)?.setMargin(2)?.qRCOde
+        qrCodeImageView.setImageBitmap(bitmap)
+    }
+
+    /**
+     * Hides the soft input keyboard if it is shown to the screen.
+     */
+
+    private fun hideKeyboard() {
```

Comment:
We already have this method implemented in `MifosBaseActivity`, you can call that only.